it/uk/gov/hmrc/agentsubscription/controllers/SubscriptionControllerISpec.scala

```diff
@@ -293,7 +293,166 @@ class SubscriptionControllerISpec extends BaseISpec with DesStubs with AuthStub
     }
   }
 
+  "updating a partial subscription" should {
+    "return a response containing the ARN" when {
+      "a valid utr is given as input" in {
+        requestIsAuthenticated().andIsAnAgent().andHasNoEnrolments()
+        agentRecordExists(utr, true, arn)
+        allocatedPrincipalEnrolmentNotExists(arn)
+        deleteKnownFactsSucceeds(arn)
+        createKnownFactsSucceeds(arn)
+        enrolmentSucceeds(groupId, arn)
+
+        val result = await(doUpdateSubscriptionRequest())
+
+        result.status shouldBe 200
+        (result.json \ "arn").as[String] shouldBe "TARN0000001"
+
+        verify(1, getRequestedFor(urlEqualTo(s"/registration/personal-details/utr/${utr.value}")))
+        verify(1, postRequestedFor(urlEqualTo(enrolmentUrl(groupId, arn))))
+      }
+    }
+
+    "return Conflict if already subscribed (both ETMP has isAsAgent=true and there is an existing HMRC-AS-AGENT enrolment for their Arn)" in {
+      requestIsAuthenticated().andIsAnAgent().andHasNoEnrolments()
```

Comment:
.andHasNoEnrolments() is correct for what were doing, but as this is a Backend, we should only check if they do not have an ARN(HMRC-AS-AGENT)?